mpbb-install-dependencies

```diff
@@ -101,34 +101,65 @@ install-dependencies() {
         text="Installing dependency ($dependencies_counter of $dependencies_count) '${depname}' with variants '${depvariants}'"
         echo "----> ${text}"
         echo -n "${text} ... " >> "$log_status_dependencies"
-        # $depvariants isn't quoted on purpose
-        # shellcheck disable=SC2086
-        if ! "${option_prefix}/bin/port" -d install --unrequested "$depname" $depvariants; then
-            echo "Build of dependency '${depname}' with variants '${depvariants}' failed, aborting." >&2
-            echo "[FAIL]" >> "$log_status_dependencies"
-            echo "Building '$port' ... [ERROR] (failed to install dependency '${depname}') maintainers: $(get-maintainers "$port" "${depname}")." >> "$log_subports_progress"
-
-            # Update failcache
-            # $depvariants isn't quoted on purpose
-            # shellcheck disable=SC2086
-            failcache_failure "$depname" $depvariants
-            if [ $? -ne 0 ]; then
-                err "failcache_failure $depname $depvariants failed."
-                return 1
-            fi
-            return 1
-        else
+        # $option_prefix and $thisdir are set in mpbb
+        # shellcheck disable=SC2154
+        if [[ -f $("${option_prefix}/bin/port-tclsh" "${thisdir}/tools/archive-path.tcl" "${depname}" "${depvariants}") ]]; then
```

Comment:
Even then you can have the case of A that depends on B that depends on C. C is broken and gets fixed at some point, we forget to rebuild B, but A gets upgraded and then builds B on the go.

I'm sure we can find plenty other cases where building dependencies will be required.

Dependency order should be fixed at some point, but I'm not sure if we can always afford to build all packages. For one, I'm not sure if PPC would **ever** catch up if we start building all packages on it. I find it way more useful to keep building the latest commits and some hand-picked packages on the build slave for a while after it gets added. OK, I guess that's enough nit-picking on dependencies :)